HLTriggerOffline/Muon/src/HLTMuonPlotter.cc

```diff
@@ -1,71 +1,79 @@
 }
 
-typedef vector<ParameterSet> Parameters;
-
-HLTMuonPlotter::HLTMuonPlotter(const ParameterSet &pset,
-                               string hltPath,
-                               const std::vector<string> &moduleLabels,
-                               const std::vector<string> &stepLabels,
-                               const edm::EDGetTokenT<trigger::TriggerEventWithRefs> &triggerEventToken,
+HLTMuonPlotter::HLTMuonPlotter(const edm::ParameterSet &pset,
+                               const std::string &hltPath,
+                               const std::vector<std::string> &moduleLabels,
+                               const std::vector<std::string> &stepLabels,
+                               const edm::EDGetTokenT<trigger::TriggerEventWithRefs> &triggerEventWithRefsToken,
                                const edm::EDGetTokenT<reco::GenParticleCollection> &genParticlesToken,
                                const edm::EDGetTokenT<reco::MuonCollection> &recoMuonsToken,
                                const L1MuonMatcherAlgoForDQM &l1Matcher)
-    : l1Matcher_(l1Matcher) {
-  hltPath_ = hltPath;
-  moduleLabels_ = moduleLabels;
-  stepLabels_ = stepLabels;
-  hltProcessName_ = pset.getParameter<string>("hltProcessName");
-
-  cutsDr_ = pset.getParameter<vector<double>>("cutsDr");
-
-  parametersEta_ = pset.getParameter<vector<double>>("parametersEta");
-  parametersPhi_ = pset.getParameter<vector<double>>("parametersPhi");
-  parametersTurnOn_ = pset.getParameter<vector<double>>("parametersTurnOn");
-
-  genMuonCut_ = pset.getParameter<string>("genMuonCut");
-  recMuonCut_ = pset.getParameter<string>("recMuonCut");
-
-  genMuonSelector_ = nullptr;
-  recMuonSelector_ = nullptr;
-
-  hltTriggerSummaryRAW_ = triggerEventToken;
-  genParticleLabel_ = genParticlesToken;
-  recMuonLabel_ = recoMuonsToken;
+    : hltPath_(hltPath),
+      hltProcessName_(pset.getParameter<std::string>("hltProcessName")),
+      moduleLabels_(moduleLabels),
+      stepLabels_(stepLabels),
+      triggerEventWithRefsToken_(triggerEventWithRefsToken),
+      genParticleToken_(genParticlesToken),
+      recMuonToken_(recoMuonsToken),
+      genMuonSelector_(pset.getParameter<std::string>("genMuonCut")),
+      recMuonSelector_(pset.getParameter<std::string>("recMuonCut")),
+      cutsDr_(pset.getParameter<std::vector<double>>("cutsDr")),
+      parametersEta_(pset.getParameter<std::vector<double>>("parametersEta")),
+      parametersPhi_(pset.getParameter<std::vector<double>>("parametersPhi")),
+      parametersTurnOn_(pset.getParameter<std::vector<double>>("parametersTurnOn")),
+      l1Matcher_(l1Matcher),
+      isInvalid_(false) {
+  if (moduleLabels_.empty()) {
+    edm::LogError("HLTMuonPlotter") << "Invalid inputs: 'moduleLabels_' is empty."
+                                    << "\nMonitorElements for HLT path '" << hltPath_ << "' will not be produced.";
+    isInvalid_ = true;
+  } else if (stepLabels_.size() != moduleLabels_.size() + 1) {
```

Comment:
This enforces the requirement that `stepLabels_.size() == moduleLabels_.size() + 1` (the difference being the first entry of `stepLabels_`, which is `"All"`, i.e. no trigger-matching requirements). This, `stepLabels_.size() == moduleLabels_.size() + 1` is guaranteed by `HLTMuonValidator::fillLabels`.